projects/samples/devices/controllers/connector/connector.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 1996-2022 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+An example of controller using a Connector device.
+"""
+
+from controller import Robot
+import math
+
+
+class Controller(Robot):
+    SPEED = 2
+    timeStep = 64
+
+    def __init__(self):
+        super(Controller, self).__init__()
+        self.connector = self.getDevice('connector')
+        self.connector.enablePresence(self.timeStep)
+        self.upper_motor = self.getDevice('upper motor')
+        self.upper_motor.setPosition(-1.57)
+        self.upper_position_sensor = self.getDevice('upper sensor')
+        self.upper_position_sensor.enable(self.timeStep)
+        self.left_motor = self.getDevice('left wheel motor')
+        self.right_motor = self.getDevice('right wheel motor')
+        self.left_motor.setPosition(float('inf'))
+        self.right_motor.setPosition(float('inf'))
+        self.left_motor.setVelocity(0.0)
+        self.right_motor.setVelocity(0.0)
+
+    def run(self):
+        robot_number = int(self.getName()[6])
+        hop_counts = 0
+        direction = 1
+        state = 'CONNECTOR_IN_PLACE'
+        left_speed = 0
+        right_speed = 0
+        while self.step(self.timeStep) != -1:
+            if state == 'CONNECTOR_IN_PLACE':  # wait for the motor to be in place
```

Comment:
```suggestion
        while self.step(self.timeStep) != -1:
            left_speed = 0
            right_speed = 0
            if state == 'CONNECTOR_IN_PLACE':  # wait for the motor to be in place
```

Otherwise the robot does not stop while rotating its arm, resulting in a clanky execution.

Reply:
Well spotted!